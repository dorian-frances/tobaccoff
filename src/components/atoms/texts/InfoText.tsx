import React from 'react';
import { Text } from 'react-native-paper';
import { StyleSheet } from 'react-native';
import { FontsEnum } from '../../../assets/fonts/fonts.enum';

export type InfoTextProps = {
  text: string;
};

const InfoText = ({ text = 'Lorem Ipsum' }: InfoTextProps) => {
  return <Text style={styles.textStyle}>{text}</Text>;
};

const styles = StyleSheet.create({
  textStyle: {
    fontFamily: FontsEnum.MEDIUM,
    fontSize: 10,
  },
});

export default InfoText;
