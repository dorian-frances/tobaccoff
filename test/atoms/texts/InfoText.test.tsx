import React from 'react';
import { faker } from '@faker-js/faker';
import { render } from '@testing-library/react-native';
import InfoText from '../../../src/components/atoms/texts/InfoText';

describe('<InfoText/>', () => {
  const fakerText = faker.datatype.string();
  it('should render correctly', () => {
    const headerText = render(<InfoText text={fakerText} />);
    expect(headerText.toJSON).toMatchSnapshot();
  });

  it('should access text', () => {
    const headerText = render(<InfoText text={fakerText} />);
    expect(headerText.getByText(fakerText)).toBeTruthy();
  });
});
