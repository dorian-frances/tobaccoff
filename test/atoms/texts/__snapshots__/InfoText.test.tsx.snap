// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoText/> should render correctly 1`] = `[Function]`;
